metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/security/MetadataAuthoritiesExtractor.java

```diff
@@ -0,0 +1,49 @@
+/*
+ *
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.metadata.security;
+
+import org.springframework.boot.autoconfigure.security.oauth2.resource.AuthoritiesExtractor;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import uk.ac.ebi.ampt2d.metadata.persistence.entities.SecurityUser;
+import uk.ac.ebi.ampt2d.metadata.persistence.repositories.SecurityUserRepository;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class MetadataAuthoritiesExtractor implements AuthoritiesExtractor {
+
+    private SecurityUserRepository securityUserRepository;
+
+    public MetadataAuthoritiesExtractor(SecurityUserRepository securityUserRepository) {
+        this.securityUserRepository = securityUserRepository;
+    }
+
+    @Override
+    public List<GrantedAuthority> extractAuthorities(Map<String, Object> map) {
+        String preferredUsername = (String) map.get("preferred_username");
+        SecurityUser securityUser = securityUserRepository.findByPreferredUsername(preferredUsername);
+        if (securityUser == null) {
+            securityUser = new SecurityUser(preferredUsername, SecurityUser.Role.ROLE_USER);
+            securityUserRepository.save(securityUser);
```

Comment:
What basically happens here is: if the user is authorised with the external server, but there is no local record for them, they are created with `ROLE_USER` and saved in the SecurityUserRepository. I have a question in this regard: wo we even _want_ to keep a track of regular users? Assuming the API is supposed to be unrestricted for **read** operations for everyone (this is my understanding anyway), maybe we don't need this role at all and can just treat any unauthorised agent as “user”?

Reply:
`can just treat any unauthorised agent as “user”?`
Others can comment if this is allowed. I followed amp-t2d-registry security style

Reply:
I guess @tskir is right we don't need to store normal users if the API is public and read by all.

Reply:
done

Reply:
At this moment i am not keeping track of regular users. ROLE_USER can be removed, however I have retained it.
Based on below comment i will to retain MetadataAuthoritiesExtractor.java or remove it entirely

Reply:
`ROLE_USER` is now removed, marking this conversation as resolved